plugins/package-managers/swiftpm/src/main/kotlin/SwiftPm.kt

```diff
@@ -159,7 +172,21 @@ class SwiftPm(
     }
 }
 
-private fun parseLockfile(packageResolvedFile: File): Set<Package> {
-    val resolved = packageResolvedFile.inputStream().use { json.decodeFromStream<PackageResolved>(it) }
-    return resolved.objects["pins"].orEmpty().mapTo(mutableSetOf()) { it.toPackage() }
+private fun parseLockfile(packageResolvedFile: File): Result<Set<Package>> {
+    val root: JsonObject = packageResolvedFile.inputStream().use { json.decodeFromStream(it) }
```

Comment:
Better use

    val root = json.parseToJsonElement(packageResolvedFile.readText()).jsonObject

to read into an "untyped" `JsonObject`.

Reply:
To also guard against JSON parsing errors, I propose to wrap the whole function body in `runCatching {}`, like

```
private fun parseLockfile(packageResolvedFile: File): Result<Set<Package>> =
    runCatching {
        val root = json.parseToJsonElement(packageResolvedFile.readText()).jsonObject

        when (val version = root.getValue("version").jsonPrimitive.content) {
            "1" -> {
                val pinsJson = root.getValue("object").jsonObject.getValue("pins")
                val pins = json.decodeFromJsonElement<List<PinV1>>(pinsJson)
                pins.mapTo(mutableSetOf()) { it.toPackage() }
            }

            else -> {
                throw IOException(
                    "Could not parse lockfile '${packageResolvedFile.invariantSeparatorsPath}'. Unknown file format " +
                        "version '$version'."
                )
            }
        }
    }
```

My code also already moves `version` to `when` to address an inspection hint that otherwise would be triggered by the next commit. And I've inlined the `messages` variable.

Reply:
I've copied your approach and only removed the assumptions that the entries for the key "object" and "pins" are always present, as I haven't found proof for this to be true.